src/drivers/imx/ipc.c

```diff
@@ -171,6 +171,10 @@ struct ipc_data_host_buffer *ipc_platform_get_host_buffer(struct ipc *ipc)
 
 int platform_ipc_init(struct ipc *ipc)
 {
+	struct task_ops ops = {
+		.run = ipc_platform_do_cmd,
+		.complete = ipc_platform_complete_cmd
+	};
```

Comment:
I don't like seeing pointers to on-stack structures being passed to functions, if the called function forgets to make a copy of the structure (this is supposedly something persistent) disaster might ensue. I may be tired but I'm not fully certain I can see whether the contents (actual function pointers) are copied or not (they probably are but I just want to make sure).

Reply:
They are copied. It's "public" API, so I don't see any problem in simple trusting the function.

Reply:
I suppose, I just considered this may be potential for future bugs (if somehow a new scheduler will be added and it doesn't account for this in its own init function).